x/dsmr/storage.go

```diff
@@ -152,33 +152,32 @@ func (s *chunkStorage[T]) SetChunkCert(chunkID ids.ID, cert *ChunkCertificate) e
 // 3. Generate a local signature share and store it in memory
 // 4. Return the local signature share
 // TODO refactor and merge with AddLocalChunkWithCert
-func (s *chunkStorage[T]) VerifyRemoteChunk(c Chunk[T]) (NoVerifyChunkSignatureShare, error) {
+func (s *chunkStorage[T]) VerifyRemoteChunk(c Chunk[T]) (*warp.BitSetSignature, error) {
 	s.lock.Lock()
 	defer s.lock.Unlock()
 
 	chunkCertInfo, ok := s.chunkMap[c.id]
 	if ok {
-		return chunkCertInfo.LocalSignature, nil
+		return chunkCertInfo.Cert.Signature, nil
 	}
 	if err := s.verifier.Verify(c); err != nil {
-		return NoVerifyChunkSignatureShare{}, err
+		return nil, err
 	}
 	if err := s.putVerifiedChunk(c, nil); err != nil {
-		return NoVerifyChunkSignatureShare{}, err
+		return nil, err
 	}
-	return NoVerifyChunkSignatureShare{}, nil
+	return nil, nil
```

Comment:
Should we be producing and returning a local signature here?

Reply:
We can remove the signature return type completely here.

From the node's perspective, when we receive a signature request we need to verify and store a chunk before returning a response, but the signature handler is now responsible for producing a signature from this node